internal/api/handlers/ledger_key_accounts.go

```diff
@@ -117,49 +131,46 @@ func processLedgerKeyAccountsEntries(publicKeys []string, data []types.LedgerEnt
 func (h *LedgerKeyAccountHandler) GetLedgerKeyAccounts(w http.ResponseWriter, r *http.Request) error {
 	contextWithTimeout, cancel := context.WithTimeout(r.Context(), HealthCheckContextTimeout)
 	defer cancel()
-	ledgerKeyAccountList := make(map[string]types.AccountInfo)
+	var ledgerKeyAccountList map[string]types.AccountInfo
 	var ledgerKeyAccountError LedgerKeyAccountError
 	queryParams := r.URL.Query()
 	network := queryParams.Get("network") 
 
-	if (len(queryParams) == 0) {
-		return httperror.BadRequest("no params passed: public key and network query params are required", errors.New("no params provided"))
-	}
-
 	if network != types.PUBLIC && network != types.TESTNET && network != types.FUTURENET {
 		return httperror.BadRequest(fmt.Sprintf("invalid network: network must be %s, %s or %s", types.PUBLIC, types.TESTNET, types.FUTURENET), errors.New("invalid network"))
 	}
 
-	for key, publicKeys := range queryParams {
-		if key == "public_key" {
-			deduplicatedPublicKeys := []string{}
-			
-			for _, publicKey := range publicKeys {
-				// RPC does not tolerate duplicate public keys, so we need to remove duplicates
-				if !slices.Contains(deduplicatedPublicKeys, publicKey) {
-					deduplicatedPublicKeys = append(deduplicatedPublicKeys, publicKey)
-				}
-			}
+	req, err := validateLedgerKeyAccountRequest(r)
+	if err != nil {
+		return httperror.BadRequest(fmt.Sprintf("%s: %s", "Invalid request - public keys are required", err.Error()), err)
+	}
 
-			ledgerKeyAccountKeys, ledgerKeyAccountKeysError := getLedgerKeyAccountKeys(deduplicatedPublicKeys)
-			if ledgerKeyAccountKeysError.ErrorMessage != "" {
-				ledgerKeyAccountError = ledgerKeyAccountKeysError
-			}
-			ledgerKeyAccountList = ledgerKeyAccountKeys.LedgerKeyAccountMap
-			
-			ledgerKeyAccountsRpcData, e := FetchLedgerEntries(h.RpcService, contextWithTimeout, ledgerKeyAccountKeys.LedgerKeys, network)
+	deduplicatedPublicKeys := []string{}
+	
+	for _, publicKey := range req.PublicKeys {
+		// RPC does not tolerate duplicate public keys, so we need to remove duplicates
+		if !slices.Contains(deduplicatedPublicKeys, publicKey) {
+			deduplicatedPublicKeys = append(deduplicatedPublicKeys, publicKey)
+		}
+	}
 
-			if e != nil && ledgerKeyAccountKeysError.ErrorMessage == "" {
-				ledgerKeyAccountError = LedgerKeyAccountError{ErrorMessage: e.Error()}
-			}
+	ledgerKeyAccountKeys, ledgerKeyAccountKeysError := getLedgerKeyAccountKeys(deduplicatedPublicKeys)
+	if ledgerKeyAccountKeysError.ErrorMessage != "" {
+		ledgerKeyAccountError = ledgerKeyAccountKeysError
+	}
+	ledgerKeyAccountList = ledgerKeyAccountKeys.LedgerKeyAccountMap
+	
+	ledgerKeyAccountsRpcData, e := FetchLedgerEntries(h.RpcService, contextWithTimeout, ledgerKeyAccountKeys.LedgerKeys, network)
```

Comment:
Shouldnt we have this function as part of the `RpcService`? See how we add the methods to the wallet backend's rpc service: https://github.com/stellar/wallet-backend/blob/main/internal/services/rpc_service.go#L207

Reply:
Good point. Actually, after iterations, `FetchLedgerEntries` is now just a simple wrapper around `RpcService.GetLedgerEntry`. I'm going to remove `FetchLedgerEntries` and just directly call `RpcService.GetLedgerEntry`